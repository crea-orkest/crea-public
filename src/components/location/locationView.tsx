import type { Location } from 'graphql/types/location'
import React from 'react'

export interface Props {
  data: Location
}

export const LocationView: React.FC<Props> = ({ data }) => {
  const { title, address, startTime } = data
  return (
    <address>
      <h2>{title}</h2>
      {address && <p>{address}</p>}
      {startTime && <p>{new Date(startTime).toDateString()}</p>}
    </address>
  )
}
