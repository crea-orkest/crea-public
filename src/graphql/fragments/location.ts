import { coordinates } from "./coordinates";
import { gql } from "@apollo/client/core";

export const location = gql`
  fragment location on LocationRecord {
    id
    title
    addressTitle
    address {
      ...coordinates
    }
  }
  ${coordinates}
`;
