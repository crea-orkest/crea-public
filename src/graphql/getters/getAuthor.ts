import type { RequestInit } from 'next/dist/server/web/spec-extension/request'
import { authorFormatter } from '../formatters/authorFormatter'
import { client } from '../gqlClient'
import {
  GetAuthorDocument,
  type GetAuthorQuery,
  type GetAuthorQueryVariables,
} from '../generated/graphql'

export const getAuthor = async (
  { id }: GetAuthorQueryVariables,
  fetchOptions?: RequestInit
) => {
  try {
    const { data, error } = await client.query<
      GetAuthorQuery,
      GetAuthorQueryVariables
    >(GetAuthorDocument, { id }, { fetchOptions })

    return {
      data: data ? authorFormatter(data) : null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)

    return { data: null, error }
  }
}
