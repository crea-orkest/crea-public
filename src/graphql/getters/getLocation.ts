import { client } from "../gqlClient";
import { locationFormatter } from "graphql/formatters/locationFormatter";
import {
  GetLocationDocument,
  type GetLocationQuery,
  type GetLocationQueryVariables,
} from "graphql/generated/graphql";

export const getLocation = async ({ id }: GetLocationQueryVariables) => {
  try {
    const { data, error } = await client.query<
      GetLocationQuery,
      GetLocationQueryVariables
    >(GetLocationDocument, {
      id,
    });

    return {
      data: data ? locationFormatter(data) : null,
      errors: error,
    };
  } catch (errors) {
    if (errors instanceof Error) console.log(errors.message);
    return { data: null, errors };
  }
};
