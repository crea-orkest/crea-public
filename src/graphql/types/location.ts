export interface Location {
  title: string;
  address: string;
  lat: number;
  lon: number;
  id: string;
}
