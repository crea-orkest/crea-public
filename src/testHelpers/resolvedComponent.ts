export const resolvedComponent = async <T>(
  Component: Function,
  props: T
): Promise<() => JSX.Element> => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};
