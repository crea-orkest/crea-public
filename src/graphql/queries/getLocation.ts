import { gql } from '@urql/core'

import { location } from 'graphql/fragments/location'

export const GET_LOCATION = gql`
  query getLocation($id: ItemId) {
    location(filter: { id: { eq: $id } }) {
      ...location
    }
  }
  ${location}
`
