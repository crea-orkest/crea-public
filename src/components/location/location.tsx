import { LocationView } from './locationView'
import React from 'react'
import { getLocation } from '../../graphql/getters/getLocation'

export interface Props {
  id: string
}

export const Location: React.FC<Props> = async ({ id }) => {
  const { data } = await getLocation({ id })
  if (!data) return null

  return <LocationView data={data} />
}
