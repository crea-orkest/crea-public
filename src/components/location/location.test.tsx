import { Location } from './location'
import type { Props } from './location'
import React from 'react'
import { getLocation } from 'graphql/getters/getLocation'
import { resolvedComponent } from '../../utils/testHelpers/resolvedComponent'
import { render, screen } from '@testing-library/react'

jest.mock('../../graphql/getters/getLocation', () => {
  const originalModule = jest.requireActual('../../graphql/getters/getLocation')
  return {
    __esModule: true,
    ...originalModule,
    getLocation: jest.fn(),
  }
})

const getLocationMock = jest.mocked(getLocation)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getLocationMock.mockResolvedValueOnce({
      data: {
        title: 'title',
        address: 'some address lines',
        lat: 5,
        lon: 42,
        id: 'mock-ids',
      },
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(Location, {
      id: 'mock-id',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('title')).toBeTruthy()
  })
})
