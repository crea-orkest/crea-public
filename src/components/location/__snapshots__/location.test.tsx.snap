// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concert component shows all the data 1`] = `
<div>
  <address>
    <h2>
      title
    </h2>
    <p>
      some address lines
    </p>
    <p>
      5
       
      42
    </p>
  </address>
</div>
`;
